packages/x-components/src/x-modules/search-box/components/__tests__/search-input-placeholder.spec.ts

```diff
@@ -0,0 +1,215 @@
+import { XPlugin } from '../../../../plugins';
+import { XEvent } from '../../../../wiring';
+import { searchBoxXModule } from '../../x-module';
+import SearchInput from '../search-input.vue';
+import SearchInputPlaceholder from '../search-input-placeholder.vue';
+
+async function mountTestComponent({
+  propsData,
+  slots
+}: TestComponentParams = {}): Promise<TestComponentAPI> {
+  const defaultMessages = ['Find sunglasses', 'Find handbags', 'Find earrings'];
+
+  const [, localVue] = installNewXPlugin();
+  XPlugin.registerXModule(searchBoxXModule);
+
+  const parent = document.createElement('div');
+  document.body.appendChild(parent);
+
+  propsData = Object.assign({ placeholder: {}, input: {} }, propsData);
+  propsData.placeholder!.messages = propsData.placeholder!.messages ?? defaultMessages;
+  propsData.input!.autofocus = propsData.input!.autofocus ?? false;
+
+  const wrapper = mount(
+    {
+      template: `
+        <div>
+        <SearchInputPlaceholder v-bind="placeholder">
+          <slot/>
+        </SearchInputPlaceholder>
+        <SearchInput v-bind="input"/>
+        </div>
+      `,
+      props: ['placeholder', 'input'],
+      components: {
+        SearchInputPlaceholder,
+        SearchInput
+      }
+    },
+    {
+      localVue,
+      /*
+       * In order to make the autofocus test work after the jest 27 update, now is mandatory to
+       * attach the element to some parent in the DOM, to emit the focus event.
+       */
+      attachTo: parent,
+      propsData: {
+        ...propsData
+      },
+      slots: {
+        ...slots
+      }
+    }
+  );
+  await wrapper.vm.$nextTick();
+
+  function getPlaceholderText(): string {
+    return wrapper.find(getDataTestSelector('search-input-placeholder')).text();
+  }
+
+  function isPlaceholderVisible(): boolean {
+    return wrapper.find(getDataTestSelector('search-input-placeholder')).exists();
+  }
+
+  function hoverInput(mode: 'in' | 'out'): Promise<void> {
+    const inputWrapper = wrapper.findComponent(SearchInput);
+    const hoverEvent = {
+      in: 'mouseenter',
+      out: 'mouseleave'
+    }[mode];
+    inputWrapper.trigger(hoverEvent);
+    return wrapper.vm.$nextTick();
+  }
+
+  async function emit(event: XEvent, payload: any): Promise<void> {
+    wrapper.vm.$x.emit(event, payload);
+    await localVue.nextTick();
+  }
+
+  return {
+    defaultMessages,
+    wrapper,
+    getPlaceholderText,
+    isPlaceholderVisible,
+    hoverInput,
+    emit
+  };
+}
+
+interface TestComponentParams {
+  propsData?: {
+    placeholder?: Partial<SearchInputPlaceholderProps>;
+    input?: SearchInputProps;
+  };
+  slots?: SearchInputPlaceholderSlots;
+}
+
+interface SearchInputPlaceholderProps {
+  messages: Array<string>;
+  animationIntervalMs?: number;
+  animateOnlyOnHover?: boolean;
+}
+
+interface SearchInputProps {
+  autofocus?: boolean;
+}
+
+interface SearchInputPlaceholderSlots {
+  default?: string;
+}
```

Comment:
Simplify this type and params:
```suggestion
interface TestComponentParams {
  messages: Array<string>;
  animationIntervalMs?: number;
  animateOnlyOnHover?: boolean;
  autofocus?: boolean;
  slotContent?: string;
}
```